roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_details.yml

```diff
@@ -0,0 +1,31 @@
+- name: Find existing VMs
+  ovirt_vms_facts:
+    auth: "{{ ovirt_auth }}"
+    pattern: "datacenter={{ data_center_name }} and name={{ hosted_engine_vm_name }}"
+    fetch_nested: true
+
+- name: Find HE info if HE
+  set_fact:
+    hosted_engine: "{{ ovirt_vms | selectattr('name', 'equalto', hosted_engine_vm_name) | map(attribute='name') | list != [] }}"
+
+- name: Find HE Storage Domain id if HE
+  block:
+    - name: Find HE VM Disk id
```

Comment:
I think you don't need this task, because it's not reliable, you should change the task below to:

```yaml
- name: Fetch whole dict for disk of HE VM
      ovirt_disks_facts:
        auth: "{{ ovirt_auth }}"
        pattern: "datacenter={{ data_center_name }} and  vm_names=cirros14 and bootable=true
```

Should you backup all disks of the HE VM or just the bootable one?

Reply:
search for bootable disk does not work but I think we don't need it anyway, I don't see use case for secondary disk for HE.